platforms/core-configuration/declarative-dsl-core/src/main/resources/org/gradle/internal/declarativedsl/parsing/grammar.bnf

```diff
@@ -43,27 +48,29 @@
 
 root ::= statements
 
-statement ::= (assignment | block | functionChain)
+statement ::= (assignment | block | expression)
 statements ::= (statement SEMI*)*
 
-assignment ::= directlyAssignableExpression (EQ|PLUSEQ) assignmentValue {pin=2}
 directlyAssignableExpression ::= simpleIdentifier navigationSuffix?
-assignmentValue ::= (simpleIdentifier (navigationSuffix* valueArguments)?) | literalOrIdentifier
+
+assignment ::= directlyAssignableExpression (EQ|PLUSEQ) expression {pin=2}
 
 block ::= simpleIdentifier valueArguments? annotatedLambda
 annotatedLambda ::= LB statements RB {pin=1}
 
-functionChain ::= simpleIdentifier postfixUnarySuffixes
-postfixUnarySuffixes ::= (navigationSuffix+ valueArguments) | (valueArguments (navigationSuffix valueArguments)*)
-
 valueArguments ::= LP (valueArgument (COMA valueArgument)* COMA?)? RP {pin=2}
-valueArgument ::= (simpleIdentifier EQ)? (functionChain | literalOrIdentifier)
+valueArgument ::= (simpleIdentifier EQ)? expression
+
+expression ::= infixFunctionCall
+infixFunctionCall ::= rangeExpression ('to' rangeExpression)?
```

Comment:
💄 Should we extract the `'to'` token into a token rule like `TO = 'to'`? Not a big deal, but for someone writing a lexer, having all the tokens extracted would be less surprising than finding more tokens inside the rules.

Reply:
If I make a token, then, in order to interpret `.to` as a navigation suffix, `navigationSuffix ::= DOT simpleIdentifier` as the rule wouldn't be enough.

Reply:
And I think it is a simple identifier, not a token, even if we don't allow any other simple identifier in its place.